include/mesh/namebased_io.h

```diff
@@ -125,8 +125,8 @@ NameBasedIO::is_parallel_file_format (std::string_view name)
 {
   return ((name.rfind(".xda") < name.size()) ||
           (name.rfind(".xdr") < name.size()) ||
-          (name.rfind(".nem") + 4 == name.size()) ||
-          (name.rfind(".n") + 2 == name.size()) ||
+          (name.rfind(".nem") == name.size() - 4) ||
+          (name.rfind(".n") == name.size() - 2) ||
```

Comment:
I'd have guessed the sanitizer wouldn't like subtracting from an unsigned type that could possibly be 0? Oh, but std::string::rfind() returns std::string::npos which is effectively max size_t so adding to that would also be bad... not sure which is best actually.